app/helpers/posts_helper.rb

```diff
@@ -0,0 +1,11 @@
+module PostsHelper
+
+  private 
+
+  def logged_in_user
+    unless logged_in? 
+      flash[:danger] = "You must log in in order to submit a post."
+      redirect_to login_url 
+    end 
+  end 
```

Comment:
If you define `logged_in?` and `current_user` in `ApplicationController`, you can expose them as available methods to the view by identifying them with the `helper_method` method.

```rb
helper_method :logged_id?, :current_user
```

See my example above, or check out the [helper_method docs](http://api.rubyonrails.org/classes/AbstractController/Helpers/ClassMethods.html#method-i-helper_method) to see where and how that's used.